### ISSUE TITLE
detect insufficient material per side

### ISSUE THREAD

**Issue by niklasf** [niklasf/shakmaty#18]
to check if a given side can still win. this is required to adjudicate positions when a player runs out of time

**Comment by niklasf** [comment]
horde not done, yet

**Comment by RoundingExplorer** [comment]
Can I help?

**Comment by niklasf** [comment]
Yes, please :)

**Comment by stevepapazis** [comment]
Whether or not the horde can mate is irrelevant; at least for this computation https://github.com/niklasf/shakmaty/blob/4b752835e7c9178bf9924cadbbddae23709f1c89/src/position.rs#L267-L269
since even a lone king preserves the winning chances for the side with the pieces. In other words, there can be no draw by insufficient material in this variant. 

So what shall `self.has_insufficient_material(White)` return? Perhaps implementing the correct behaviour is not worth the trouble.

**Comment by niklasf** [comment]
Correct, a Horde game can never end by `is_insufficient_material`. But Black could run out of time, in a position where `self.has_insufficient_material(White)` is true. Then the game should be drawn, not lost for Black. [Current docs](https://docs.rs/shakmaty/0.17.1/shakmaty/trait.Position.html#tymethod.has_insufficient_material) explain the meaning.
